samples/snippets/src/main/java/com/examples/documentai/ParseForm.java

```diff
@@ -0,0 +1,101 @@
+
+public class ParseForm {
+  public static void parseForm() throws IOException, ExecutionException, InterruptedException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String location = "your-region";    // available regions https://cloud.google.com/compute/docs/regions-zones
+    String inputGcsUri = "gs://your-gcs-bucket/path/to/input/file.json";
+    parseForm(projectId, location, inputGcsUri);
+  }
+
+  public static void parseForm(String projectId, String location, String inputGcsUri)
+      throws IOException, ExecutionException, InterruptedException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (DocumentUnderstandingServiceClient client = DocumentUnderstandingServiceClient.create()) {
+      // Configure the request for processing the PDF
+      String parent = String.format("projects/%s/locations/%s", projectId, location);
+
+      KeyValuePairHint keyValuePairHint =
+          KeyValuePairHint.newBuilder().setKey("Phone").addValueTypes("PHONE_NUMBER").build();
+      KeyValuePairHint keyValuePairHint2 =
+          KeyValuePairHint.newBuilder()
+              .setKey("Contact")
+              .addValueTypes("EMAIL")
+              .addValueTypes("NAME")
+              .build();
+
+      FormExtractionParams params =
+          FormExtractionParams.newBuilder()
+              .setEnabled(true)
+              .addKeyValuePairHints(keyValuePairHint)
+              .addKeyValuePairHints(keyValuePairHint2)
+              .build();
+
+      GcsSource uri = GcsSource.newBuilder().setUri(inputGcsUri).build();
+
+      InputConfig config =
+          InputConfig.newBuilder().setGcsSource(uri).setMimeType("application/pdf").build();
+
+      ProcessDocumentRequest req =
+          ProcessDocumentRequest.newBuilder()
+              .setParent(parent)
+              .setFormExtractionParams(params)
+              .setInputConfig(config)
+              .build();
+
+      // Recognizes text entities in the PDF document
+      Document res = client.processDocument(req);
+
+      // Get all of the document text as one big string
+      String text = res.getText();
+
+      // Process the output
+      Document.Page page1 = res.getPages(0);
+      for (Document.Page.FormField field : page1.getFormFieldsList()) {
+        String fieldName = getText(field.getFieldName().getTextAnchor(), text);
+        String fieldValue = getText(field.getFieldValue().getTextAnchor(), text);
+
+        System.out.println("Extracted key value pair:");
+        System.out.printf("\t(%s, %s))", fieldName, fieldValue);
+      }
+    }
+  }
+
+  private static String getText(Document.TextAnchor textAnchor, String text) {
```

Comment:
Use `Document.Layout` that way each time you call this method you don't have to call `getTextAnchor()`